Fix, refactor, move xml_interface

peterschrammel opened the pull request:
- XML_UI was not enabled
- Turn the class into a free function
- Move to xmllang. Enables reuse in other driver programs. It's not CBMC-specific.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

peterschrammel committed:
Move xml_interface to xmllang

Enables reuse in other driver programs.
It's not CBMC-specific.

tautschnig reviewed (commented):
Facilitating reuse seems like a good idea, but:
1) All CI is failing.
2) The class lacks any documentation, thus I'm not sure "reuse" is actually possible? If you have figured out how to (re-)use it, could you please contribute documentation?
3) The implementation uses `null_message_handlert`, which raises alarm bells as there is a risk of hiding errors. I think it should just take a `message_handlert` as an argument.
4) There is a class, but really its only public member is a constructor. Using a free function may be much better and easier, with the two implementation methods just becoming static functions.

peterschrammel committed:
Bugfix: Use XML_UI if xml-interface is enabled

This feature was broken by a previous parse_optionst
refactoring. We can either enable XML_UI if xml-interface
is enabled or give ui_message_handlert a set_ui() method
to set the UI after cmdline has been fully parsed.

peterschrammel committed:
Refactor xml_interface

Turn the constructor-only class into a free function.

peterschrammel committed:
Add xml-interface test

peterschrammel committed:
Clang-format
